addons/quest_system/quest_manager.gd

```diff
@@ -147,33 +146,61 @@ func call_quest_method(quest_id: int, method: String, args: Array) -> void:
 
 
 func set_quest_property(quest_id: int, property: String, value: Variant) -> void:
-	var quest: Quest = null
-
+	
 	# Find the quest
-	for pools in get_all_pools():
-		if pools.get_quest_from_id(quest_id) != null:
-			quest = pools.get_quest_from_id(quest_id)
+	var quest: Quest = __get_quest_by_id(quest_id)
 
 	if quest == null: return
 
 	# Now check if the quest has the property
+	var property_found : bool = __quest_has_property(quest, property)
+
+	# Return if the property was not found
+	if not property_found: return
 
-	# First if the property is null -> we return
-	if property == null: return
+	# Finally we set the value
+	quest.set(property, value)
+
+func get_quest_property(quest_id: int, property: String) -> Variant:
```

Comment:
This API exposed method should be documented in `/docs/api/quest_manager.md` too

Reply:
Oh, so that what this means in the checklist: ``` I've properly updated the documentation to reflect the changes I've made.```

Sorry, my bad, I will fix this